app/src/main/java/org/schabi/newpipe/fragments/list/playlist/PlaylistFragment.java

```diff
@@ -267,6 +270,17 @@ public boolean onOptionsItemSelected(final MenuItem item) {
             case R.id.menu_item_bookmark:
                 onBookmarkClicked();
                 break;
+            case R.id.menu_item_append_playlist:
+                disposables.add(PlaylistDialog.createCorrespondingDialog(
+                        getContext(),
+                        getPlayQueue()
+                                .getStreams()
```

Comment:
The issue with this is that if the playlist was not completely loaded, only the first few items are added to the other playlist, not all of them. E.g. my YouTube music playlist has ~200 videos, but YouTube loads 100 items at once, so in order to add the whole playlist to another playlist with this option I'd first have to scroll down to the bottom so that I'm sure it was loaded completely.

Reply:
I had tried having it load the entire playlist automatically when selecting the menu option. However, the lack of a loading indicator when loading large playlists may be a problem (see video). What are your thoughts?

https://user-images.githubusercontent.com/88139840/144722928-d10e717a-18e3-4897-bf7f-7f24ecba7053.mp4

Reply:
I'm not sure. Somebody will definitely find this useful, but I wouldn't introduce full playlist loading in this PR since it introduces some things that need to be resolved, and I would address them in a separate PR. You'd be welcome to create such a PR :-)
- a loading indicator should be shown somewhere, as you said
- what happens if the playlist length is infinite (and we could not know that it is infinite)? Infinite playlists are e.g. YouTube mixes. We have to discuss this.

What do you think @petlyh @litetex @TobiGr @TiA4f8R ?